apps/admin_app/assets/js/components/CreatePromotions.tsx

```diff
@@ -78,43 +81,52 @@ export class PromotionsForm extends React.Component<any, any>{
         }
 
         if (this.props.editResponse === undefined) {
-            const url = myconstants.PROMOTIONS_LIST_URL;
-            fetchPost(url, data).then(res => res.json()).then(response => {
-                var arrofkeys = Object.keys(response)
-                arrofkeys[0] == "errors" ? this.onFail(response) : this.onSuccess()
+            const PromotionsListURL = PromotionsConstants.PROMOTIONS_LIST_URL;
+            fetchPost(PromotionsListURL, data).then(res => {
+                if(res.status==422){
+                    return this.onFail(res.json())
+                }
+                else if(res.ok){
+                    return this.onSuccess()
+                }
             })
-                .catch(error => console.error('Error', error));
+            .catch(error => console.error('Error', error));
         }
         else {
             let id = this.props.editResponse["attributes"]["id"]
-            const url = myconstants.PROMOTIONS_LIST_URL + id
-
-            fetchPut(url, data).then(res => res.json()).then(response => {
-                var arrofkeys = Object.keys(response)
-                arrofkeys[0] == "errors" ? this.onFail(response) : this.onSuccess()
-            }).catch(error => console.error('Error', error))
+            const PromotionsUpdateURL = PromotionsConstants.PROMOTIONS_LIST_URL + id
+
+            fetchPut(PromotionsUpdateURL, data).then(res => {
+                if(res.status==422){
+                    return this.onFail(res.json())
+                }
+                else if(res.ok){
+                    return this.onSuccess()
+                }
+            })       
+            .catch(error => console.error('Error', error))
         }
     }
 
     onSuccess = () => {
         this.setState({ success: true })
     }
-    onFail = (response) => {
-        this.setState({ failed: true, errors: response })
-        Object.keys(response).map((res) => {
-            var errormsg = ""
-            Object.keys(response[res]).map((res1) => {
-                errormsg = errormsg + res1 + " " + response[res][res1][0]["message"] + "\n"
-            })
+
+    onFail = (res) => {
+        res.then(response=>{
+            this.setState({ failed: true, errors: response })
         })
+
     }
+
     handleAddRule = () => {
         this.setState({ addRule: true })
-        const url = myconstants.RULES_LIST_URL;
-        fetchGet(url).then(res => res.json()).then(response => {
-            var availableRules = response
+        const RulesListURL = PromotionsConstants.RULES_LIST_URL;
+        fetchGet(RulesListURL).then(res => res.json()).then(response => {
+            let availableRules = response
             this.setState({ availableRules: availableRules })
         })
+        this.setState({availableProducts:fetchProducts()})
```

Comment:
space between

Reply:
@SakinaBoriwala21 Please add
```
yarn add prettier --dev or npm install prettier --save-dev
```

Then in your vscode create a folder .vscode
```
{
  "editor.fontSize": 12,
  "editor.formatOnPaste": true,
  "editor.formatOnSave": true,
  "prettier.eslintIntegration": true,
  "files.insertFinalNewline": true
}
```

Reply:
This will help you avoid all the above styling and spacing issues

Reply:
@VivekNayyar Thanks a lot Sir!